### PR TITLE
Added ACMEv2 classes, ACMEDatabase, ACMEBackend, and ACMEEngine

### PR DESCRIPTION
**Pull request by edewata**
See commit description for more info.

The complete set of patches is available at:
https://github.com/edewata/pki/commits/acme-dev

### CODE REVIEW

File: base/acme/src/org/dogtagpki/acme/ACMEHeader.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+/**
+ * @author Endi S. Dewata
+ */
+@JsonInclude(Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class ACMEHeader {
+
+    private String alg;
+    private JWK jwk;
```
Comment (cipherboy) [inline base/acme/src/org/dogtagpki/acme/ACMEHeader.java]: UGH, JWK/JWS is terrible to get right. Do you have that implemented yet? Are you using a helper library for it?

Comment (edewata) [reply base/acme/src/org/dogtagpki/acme/ACMEHeader.java]: In a later patch I'm generating JWK thumbprint to be used as account ID which is I think also used for ACME challenge. So far it's working fine with certbot. See the following code:
* https://github.com/edewata/pki/commit/cbc823aecc3462132cae3f1f2826ace34372c372#diff-a3185bddea60201699a80824778a9af4R50
* https://github.com/edewata/pki/commit/cbc823aecc3462132cae3f1f2826ace34372c372#diff-d9d69448fbf354ff2619a045507ec743R245

Comment (cipherboy) [reply base/acme/src/org/dogtagpki/acme/ACMEHeader.java]: Hmm ok. Thanks!

**Comment by edewata** [comment]
Thanks!

**Pull request merged**